substrafl/strategies/fed_pca.py

```diff
@@ -0,0 +1,117 @@
+from typing import List
+from typing import Optional
+
+import numpy as np
+from numpy import linalg
+
+from substrafl.algorithms.algo import Algo
+from substrafl.exceptions import EmptySharedStatesError
+from substrafl.nodes.aggregation_node import AggregationNode
+from substrafl.nodes.references.local_state import LocalStateRef
+from substrafl.nodes.references.shared_state import SharedStateRef
+from substrafl.nodes.train_data_node import TrainDataNode
+from substrafl.remote import remote
+from substrafl.schemas import FedAvgAveragedState
+from substrafl.schemas import FedAvgSharedState
+from substrafl.schemas import StrategyName
+from substrafl.strategies import FedAvg
+
+
+class FedPCA(FedAvg):
+    """Federated PCA strategy."""
+
+    def __init__(self):
+        super(FedPCA, self).__init__()
+
+        # current local and share states references of the client
+        self._local_states: Optional[List[LocalStateRef]] = None
+        self._shared_states: Optional[List[SharedStateRef]] = None
+
+    @property
+    def name(self) -> StrategyName:
+        """The name of the strategy
+
+        Returns:
+            StrategyName: Name of the strategy
+        """
+        return StrategyName.FEDERATED_PCA
+
+    def perform_round(
+        self,
+        algo: Algo,
+        train_data_nodes: List[TrainDataNode],
+        aggregation_node: AggregationNode,
+        round_idx: int,
+        clean_models: bool,
+        additional_orgs_permissions: Optional[set] = None,
+    ):
+        """One round of the Federated PCA"""
+        if aggregation_node is None:
+            raise ValueError("In FedAvg strategy aggregation node cannot be None")
+
+        if round_idx == 0:
+            # Initialization of the strategy by performing a local update on each train data organization
+            assert self._local_states is None
+            assert self._shared_states is None
+            self._perform_local_updates(
+                algo=algo,
+                train_data_nodes=train_data_nodes,
+                current_aggregation=None,
+                round_idx=round_idx,
+                aggregation_id=aggregation_node.organization_id,
+                additional_orgs_permissions=additional_orgs_permissions or set(),
```

Comment:
```suggestion
                additional_orgs_permissions=additional_orgs_permissions or {},
```
Using directly an empty set is faster than calling the `set` constructor with no argument.

Reply:
`{}` is considered as an empty dict and not a set. This is a case where we need the `set()` IMO.

Reply:
Indeed, my bad